Source/VMware.PSDesiredStateConfiguration/Classes/DscItems.ps1

```diff
@@ -1231,6 +1225,57 @@ class DscConfigurationRunner {
         return $invokeResult
     }
 
+    <#
+    .DESCRIPTION
+    Prints the output from the execution of a DSC Resource via the Invoke-DscResource cmdlet.
+    The logs are extracted from a file, located in the Temp directory of the OS, where the execution occurred.
+    This extraction and printing is required, because the streams are not available during the execution of the DSC Resource.
+    #>
+    hidden [void] HandleStreamOutputFromDscResources([VmwDscResource] $DscResource) {
+        # The Temp folder is not available when executing the build procedure with Travis CI.
+        if ($null -eq $env:TEMP) {
+            return
+        }
+
+        $logsToExtract = @('Warning', 'Verbose')
+
+        foreach ($logType in $logsToExtract) {
+            # get stream preference
+            # note: preference 
+            $logPreference = (Get-Variable -Name ($logType + 'Preference')).Value
+
+            if ($logPreference -ne 'Continue') {
+                continue
+            }
+
+            $connectionName = [string]::Empty
+
+            if ($DscResource.Property.ContainsKey('Connection')) {
+                $connectionName = $DscResource.Property['Connection'].Name
+            } elseif ($DscResource.Property.ContainsKey('Server')) {
+                $connectionName = $DscResource.Property['Server']
+            }
+
+            $resourceName = $DscResource.ResourceType
```

Comment:
Here I think we will have a name collision - for example a **Configuration** consisting of two **Datacenter DSC Resources** - the second one executed will override the log of the first one as **ResourceType** is not unique. To differentiate between two instances of a **DSC Resource**, the ideal scenario is to take in mind the key properties of each instance of a **DSC Resource**. However this can't be implemented in the base class as the current implementation does. One way to achieve uniqueness is to define a new hidden property to the Base class of type **GUID** - something like client id, which will be unique for each instance of a DSC Resource. This can then be added to the **VmwDscResource** class as a property.

With that in mind, this is not a trivial change and it should be implemented as a separate PR as the current one already changes a lot.

For now leave it like this and we can discuss it more before implementing it.

Reply:
I changed the logic so that a shared variable is used and files are no longer required.